ios/TPSTwitterModule/TPSTwitterModule.m

```diff
@@ -0,0 +1,222 @@
+#import "TPSTwitterModule.h"
+#import <TwitterKit/TwitterKit.h>
+#import <Social/Social.h>
+#import "RCTUtils.h"
+#import <OAuthCore.h>
+
+typedef void(^TWTAPIHandler)(NSData *data, NSError *error);
+typedef void(^TWTRSessionHandler)(TWTRSession *session, NSError *error);
+static NSString *const kTWTConsumerKey = @"consumerKey";
+static NSString *const kTWTConsumerSecret = @"consumerSecret";
+
+@interface TPSTwitterModule ()
+@property (nonatomic, copy) NSString *consumerKey;
+@property (nonatomic, copy) NSString *consumerSecret;
+@end
+
+@implementation TPSTwitterModule
+
+RCT_EXPORT_MODULE()
+
+- (dispatch_queue_t)methodQueue {
+    return dispatch_get_main_queue();
+}
+
+RCT_EXPORT_METHOD(init:(NSDictionary*)twitterCredentials
+                  resolve:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject) {
+    if (twitterCredentials[kTWTConsumerKey] && twitterCredentials[kTWTConsumerSecret]) {
```

Comment:
The best way is create local variables for twitterCredentials[kTWTConsumerKey] and twitterCredentials[kTWTConsumerSecret] for using in this method